plugins/modules/repository_set_info.py

```diff
@@ -0,0 +1,166 @@
+      - Some repository sets require only I(basearch) or only I(releasever)
+      - Required when I(all_repositories) is unset or C(false)
+    required: false
+    type: list
+    elements: dict
+    suboptions:
+      basearch:
+        description:
+          - Basearch of the repository
+        type: str
+      releasever:
+        description:
+          - Releasever of the repository
+        type: str
+  all_repositories:
+    description:
+      - Find all repositories in the repository set
+      - Required when I(repositories) is unset or an empty list
+    required: false
+    type: bool
+  enabled:
+    description:
+      - When true, only return repository sets which are already enabled. Defaults to false.
+    required: false
+    default: false
+    type: bool
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.katelloinfomodule
+  - theforeman.foreman.foreman.infomodule
+'''
+
+EXAMPLES = '''
+- name: "Find repository set by name and product. Match repositories with the given releasever and basearch only"
+  theforeman.foreman.repository_set_info:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    organization: "Default Organization"
+    name: "Red Hat Enterprise Linux 7 Server (RPMs)"
+    product: "Red Hat Enterprise Linux Server"
+    repositories:
+      - releasever: "7Server"
+        basearch: "x86_64"
+      - releasever: "7.9"
+        basearch: "x86_64"
+
+- name: "Find repository set by name and product. Match all repositories in the set"
+  theforeman.foreman.repository_set_info:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    organization: "Default Organization"
+    name: "Red Hat Enterprise Linux 7 Server (RPMs)"
+    product: "Red Hat Enterprise Linux Server"
+    all_repositories: true
+
+- name: "Find repository set by label. Match all enabled repositories in the set"
+  theforeman.foreman.repository_set_info:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    organization: "Default Organization"
+    label: rhel-7-server-rpms
+    all_repositories: true
+    enabled: true
+'''
+
+RETURN = '''
+repository_sets:
+  description: List of repository sets.
+  returned: success
+  type: list
+  elements: dict
+'''
+
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import KatelloInfoAnsibleModule
+
+
+class KatelloRepositorySetInfo(KatelloInfoAnsibleModule):
+    pass
+
+
+def main():
+    module = KatelloRepositorySetInfo(
+        foreman_spec=dict(
+            product=dict(type='entity', scope=['organization']),
+            name=dict()
+            label=dict()
+            repositories=dict(type='list', elements='dict', options=dict(
+                basearch=dict(),
+                releasever=dict(),
+            )),
+            all_repositories=dict(type='bool'),
```

Comment:
I don't see any code using `repositories` and/or `all_repositories` and would argue that these are totally unnecessary in this info module, as as a user, I want to fetch information about the *whole* repo set, and see which repos of it are enabled and which not.